layers/+email/mu4e/packages.el

```diff
@@ -38,4 +39,10 @@
 
       (when mu4e-account-alist
         (add-hook 'mu4e-compose-pre-hook 'mu4e/set-account)
-        (add-hook 'message-sent-hook 'mu4e/mail-account-reset)))))
+        (add-hook 'message-sent-hook 'mu4e/mail-account-reset))
+      (if mu4e-use-maildirs
+        (mu4e-maildirs-extension)))))
+
+(defun mu4e/init-mu4e-maildirs-extension ()
+  (use-package mu4e-maildirs-extension
+    :commands (mu4e-maildirs-extension)))
```

Comment:
Layer variables should be set together with the layer declaration in the dotfile, so they ought to be set at this point. I agree with @StreakyCobra that you can use `:if` here, and move the call to `mu4e-maildors-extension` as well (use `with-eval-load` or similar).